common/djangoapps/microsite_configuration/backends/database.py

```diff
@@ -0,0 +1,164 @@
+
+    def get_all_orgs(self):
+        """
+        This returns a set of orgs that are considered within a microsite. This can be used,
+        for example, to do filtering
+        """
+
+        # This should be cacheable (via memcache to keep consistent across a cluster)
+        # I believe this is called on the dashboard and catalog pages, so it'd be good to optimize
+        return set(MicrositeOrgMapping.objects.all().values_list('org', flat=True))
+
+    def _set_microsite_config_from_obj(self, subdomain, domain, microsite_object):
+        """
+        Helper internal method to actually find the microsite configuration
+        """
+        config = json.loads(microsite_object.values)
+        config['subdomain'] = subdomain
+        config['site_domain'] = domain
+        config['microsite_config_key'] = microsite_object.key
+
+        # we take the list of ORGs associated with this microsite from the database mapping
+        # tables. NOTE, for now, we assume one ORG per microsite
+        orgs = microsite_object.get_orgs()
+
+        # we must have at least one ORG defined
+        if not orgs:
+            raise Exception(
+                'Configuration error. Microsite {key} does not have any ORGs mapped to it!'.format(
+                    key=microsite_object.key
+                )
+            )
+
+        # just take the first one for now, we'll have to change the upstream logic to allow
+        # for more than one ORG binding
+        config['course_org_filter'] = orgs[0]
+        self.current_request_configuration.data = config
+
+
+class DatabaseMicrositeTemplateBackend(BaseMicrositeTemplateBackend):
+    """
+    Specialized class to pull templates from the database
+    """
+    def get_template_path(self, relative_path, **kwargs):
+        return relative_path
+
+    def get_template(self, uri):
+        """
+        Override of the base class for us to look into the
+        database tables for a template definition, if we can't find
+        one we'll return None which means "use default means" (aka filesystem)
+        """
+        cache_key = "template_cache." + microsite_get_value('microsite_config_key') + '.' + uri
+        template_text = cache.get(cache_key)  # pylint: disable=maybe-no-member
+
+        if not template_text:
+            # Don't use the cache
+            template_obj = MicrositeTemplate.get_template_for_microsite(
+                microsite_get_value('microsite_config_key'),
+                uri
+            )
+
+            if not template_obj:
+                # We need to set something in the cache to improve performance
+                # of the templates stored in the filesystem as well
+                cache.set(cache_key, '##none', 60 * 5)  # pylint: disable=maybe-no-member
+                return None
+
+            template_text = template_obj.template
+            cache.set(cache_key, template_text, 60 * 3)  # pylint: disable=maybe-no-member
```

Comment:
Ahhh, so that is why you are storing '##none' rather than deleting the cached object. That is so it will be a 'quick detection' when we know there is no underlying object.

How are we invalidating the cached '##none'? I did a quick look to see if on the model.py we are deleting the cached object and I didn't see anything.

Did I miss something, while the '##none' placeholder seems like a good idea we have to make sure that it accurately reflects what is in the underlying database....